### ISSUE TITLE
Scope table not listing items

### ISSUE THREAD

**Issue by burtonc** [pwndoc-ng/pwndoc-ng#121]
**Describe the bug**
When reloading audit, scope items shows as `[Object object]`.

**To Reproduce**
1. Create an audit
2. Add data in "scope" field
3. Reload page

**Expected behavior**
Items filled in "scope" field should display their name instead of `[Object object]`.

**Screenshots**
![image](https://user-images.githubusercontent.com/24873615/228021956-b2192172-cad8-412c-8ed7-bd602719cbfb.png)

**Environment (please complete the following information):**
 - Tested on Firefox / Chrome

**Additional context**
Scope name display well in network view. It use to work well in previous version of pwndoc-ng.

**Comment by Zeecka** [comment]
I can reproduce, I've got the same issue with original pwndoc too (I don't know if you can confirm).
This is a front-end issue only (it should display correctly in the repport). I'm looking for a fix but I'm not a Vue.js expert :') !

**Comment by burtonc** [comment]
This issues has only recently appeared in my install, now running pwndoc-ng, original pwndoc install about 6 to 8 weeks old is not impacted

upon investigation the export to word shows the scope despite there UI not displaying it.

I’ve proxied the calls, there are 2 API responses returned on the page, first one responds wil scope correctly, second responds empty for scope hosts (I’ll update with my requests later as not in front of that machine)

**Comment by Zeecka** [comment]
It looks like we have a regression on the API side (I think it's due to a version bump). I'm currently checking if data is altered before insert or on query.

### Test on API route
 `api/audits/<id>/general`

#### Working version
```json
{
    "status": "success",
    "datas": {
        "collaborators": [],
        "reviewers": [],
        "_id": "6421fdbfc56624001175598f",
        "name": "Test",
        "language": "fr",
        "auditType": "Web",
        "creator": {
            "_id": "6421fd80c56624001175598a",
            "username": "admin",
            "firstname": "admin",
            "lastname": "admin"
        },
        "customFields": [],
        "template": "6421fd8cc56624001175598c",
        "scope": [
            "aa",
            "bbb"
        ]
    }
}
```

#### Last version
```json
{
    "status": "success",
    "datas": {
        "_id": "6421ff769522ee7a0971f8ae",
        "name": "fr",
        "auditType": "fr",
        "collaborators": [],
        "reviewers": [],
        "language": "fr",
        "template": "6421ff519522ee7a0971f877",
        "creator": {
            "_id": "6421ff379522ee7a0971f861",
            "username": "admin",
            "firstname": "admin",
            "lastname": "admin"
        },
        "customFields": [],
        "scope": [
            {
                "hosts": []
            },
            {
                "hosts": []
            }
        ]
    }
}
```

**Comment by Zeecka** [comment]
DB entry is the same on both version (which is coherent with the fact report generation and network section are working).

```
{
    "_id": ObjectId("64220c47778ed0b87cb5f705"),
    "name": "a",
    "auditType": "fr",
    "collaborators": [],
    "reviewers": [],
    "language": "fr",
    "template": ObjectId("64220c37778ed0b87cb5f6ed"),
    "creator": ObjectId("64220c2d778ed0b87cb5f6d7"),
    "sections": [],
    "customFields": [],
    "sortFindings": [],
    "state": "EDIT",
    "approvals": [],
    "scope": [
        {
            "name": "aa",
            "hosts": []
        },
        {
            "name": "bbb",
            "hosts": []
        }
    ],
    "findings": [],
    "createdAt": ISODate("2023-03-27T21:36:07.524Z"),
    "updatedAt": ISODate("2023-03-27T21:36:11.768Z"),
    "__v": 0
}
```

**Comment by Zeecka** [comment]
It may be related to mongoose upgrade ( see https://github.com/pwndoc/pwndoc/commit/af758090b04aa039170247b3d5c4fb243ac80161#diff-eb434d797b3f5637e7269f5bc9be8842a8e8761c93f3f8347326f9418c746b8cL28 )

**Comment by diggidong** [comment]
Unfortunately this does not fix it on my side.
I am on the latest pwndoc-ng master branch.
It renders right when exporting,
but the web app still does have issues with it.
If I reopen a report and it shows [object Object], I am also not able to save it, because of the error:
`Cast to embedded failed for value "{ name: { hosts: [] } }" (type Object) at path "scope" because of "CastError"`

![image](https://user-images.githubusercontent.com/9928172/228425869-a3ad2adb-f8fb-400e-bff3-1ffe234b38df.png)

**Comment by Zeecka** [comment]
Fixed with #135
